src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/References/ReferenceCleanupService.ReferenceHandlers.cs

```diff
@@ -0,0 +1,185 @@
+                Requires.NotNull(configuredProject, nameof(configuredProject));
+                Assumes.Present(configuredProject.Services);
+
+                return RemoveReferenceAsync(configuredProject.Services, reference);
+            }
+
+            protected abstract Task RemoveReferenceAsync(ConfiguredProjectServices services,
+                ProjectSystemReferenceInfo referencesInfo);
+
+            private Task<IEnumerable<IProjectItem>> GetUnresolvedReferencesAsync(ConfiguredProject selectedConfiguredProject)
+            {
+                Requires.NotNull(selectedConfiguredProject, nameof(selectedConfiguredProject));
+                Assumes.Present(selectedConfiguredProject.Services);
+
+                return GetUnresolvedReferencesAsync(selectedConfiguredProject.Services);
+            }
+
+            protected abstract Task<IEnumerable<IProjectItem>> GetUnresolvedReferencesAsync(ConfiguredProjectServices services);
+
+            internal async Task<bool> CanRemoveReferenceAsync(ConfiguredProject selectedConfiguredProject, ProjectSystemReferenceUpdate referenceUpdate, CancellationToken cancellationToken)
+            {
+                var references = await GetReferencesAsync(selectedConfiguredProject, cancellationToken);
+
+                ProjectSystemReferenceInfo referenceInfo = references.FirstOrDefault(c => c.ItemSpecification == referenceUpdate.ReferenceInfo.ItemSpecification);
+
+                if (referenceInfo is null)
+                {
+                    return false;
+                }
+
+                return referenceInfo.TreatAsUsed != true;
+            }
+
+            internal async Task<List<ProjectSystemReferenceInfo>> GetReferencesAsync(ConfiguredProject selectedConfiguredProject, CancellationToken cancellationToken)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var references = new List<ProjectSystemReferenceInfo>();
+
+                var projectItems = await GetUnresolvedReferencesAsync(selectedConfiguredProject);
+
+                if (projectItems is null)
+                {
+                    return references;
+                }
+
+                foreach (var item in projectItems)
+                {
+                    string? treatAsUsed = await GetAttributeTreatAsUsedAsync(item.Metadata);
+                    string? itemSpecification = item.UnevaluatedInclude;
+
+                    if (!string.IsNullOrEmpty(itemSpecification))
+                    {
+                        references.Add(new ProjectSystemReferenceInfo(_referenceType, itemSpecification, treatAsUsed == bool.TrueString));
+                    }
+                }
+
+                return references;
+            }
+
+            private static async Task<string?> GetAttributeTreatAsUsedAsync(IProjectProperties metadata)
+            {
+                string? value = await metadata.GetUnevaluatedPropertyValueAsync(ProjectReference.TreatAsUsedProperty);
```

Comment:
This must use "Evaluated" not unevaluated.

Reply:
done